File: backend/src/main/java/com/cdac/farmermarketplace/service/WishlistService.java
package com.cdac.farmermarketplace.service;

import java.util.List;

import com.cdac.farmermarketplace.dto.response.WishlistResponseDto;


public interface WishlistService {


    // Add product to wishlist
    WishlistResponseDto addToWishlist(
            Long productId,
            String email
    );


    // Get logged-in user's wishlist
    List<WishlistResponseDto> getWishlist(
            String email
    );


    // Remove product from wishlist
    void removeFromWishlist(
            Long productId,
            String email
    );

}

File: frontend/src/services/wishlistService.js
import api from "./api";

const wishlistService = {
  getWishlist: async () => {
    const email = localStorage.getItem("email");

    const response = await api.get("/wishlist", {
      params: { email },
    });

    return response.data;
  },

  addToWishlist: async (productId) => {
    const email = localStorage.getItem("email");

    const response = await api.post(
      `/wishlist/${productId}`,
      null,
      {
        params: { email },
      }
    );

    return response.data;
  },

  removeFromWishlist: async (productId) => {
    const email = localStorage.getItem("email");

    const response = await api.delete(
      `/wishlist/${productId}`,
      {
        params: { email },
      }
    );

    return response.data;
  },
};

export default wishlistService;